dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -691,10 +721,70 @@ public Workflow getWorkflowByPath(@ApiParam(hidden = true) @Auth User user, @Api
 
         Workflow workflow = workflowDAO.findByPath(path, false);
         checkEntry(workflow);
-        checkUser(user, workflow);
+        try {
+            checkUser(user, workflow);
+        } catch (CustomWebApplicationException ex) {
+            if (!permissionsInterface.canDoAction(user, workflow, Action.READ)) {
+                throw ex;
+            }
+        }
         return workflow;
     }
 
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/path/workflow/{repository}/permissions")
+    @ApiOperation(value = "Get all permissions for a workflow", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, notes = "Lists all permissions for a workflow. The user must be the workflow owner.", response = Permission.class, responseContainer = "List")
+    public List<Permission> getWorkflowPermissions(@ApiParam(hidden = true) @Auth User user, @ApiParam(value = "repository path", required = true) @PathParam("repository") String path) {
+        Workflow workflow = workflowDAO.findByPath(path, false);
+        checkEntry(workflow);
+        return this.permissionsInterface.getPermissionsForWorkflow(user, workflow);
+    }
+
+    @PATCH
```

Comment:
Could this just be a Put since it seems idempotent

Reply:
I made it PATCH because it's only modifying a part of the collection. For example, if the permissions on a resource are:
```
[
  {
    "email": "jdoe@ucsc.edu",
    "role": "OWNER"
  },
  {
    "email": "jane.doe@gmail.com",
    "role": "READER"
  }
]
```
Then I could add another user without having to specify the whole body (and that would less likely for users to clobber each other's changes).